tests/test_facebook_all_fields.py

```diff
@@ -0,0 +1,31 @@
+"""
+Test that with no fields selected for a stream all fields are still replicated
+"""
+
+from tap_tester.base_suite_tests.all_fields_test import AllFieldsTest
+from base_new_frmwrk import FacebookBaseTest
+import base
+
+
+class FacebookAllFieldsTest(AllFieldsTest, FacebookBaseTest):
+    """Test that with no fields selected for a stream all fields are still replicated"""
+    is_done = None
+
+    @staticmethod
+    def name():
+        return "tt_facebook_all_fields_test"
+
+    def streams_to_test(self):
+        #return set(self.expected_metadata().keys())
+        # Fail the test when the JIRA card is done to allow stream to be re-added and tested
+        if self.is_done is None:
+            self.is_done = base.JIRA_CLIENT.get_status_category("TDL-24312") == 'done'
+            self.assert_message = ("JIRA ticket has moved to done, re-add the "
+                                   "ads_insights_hourly_advertiser stream to the test.")
+        assert self.is_done != True, self.assert_message
+
+        return self.expected_metadata().keys() - {'ads_insights_hourly_advertiser', 'ads_insights'}
+
+    def test_all_fields_for_streams_are_replicated(self):
+        self.selected_fields = self.get_upsert_only_fields(AllFieldsTest.selected_fields)
```

Comment:
Looks good.  My only concern is making sure that get_upsert_only_fields() stays up to date.  Can we come up with a way to test that in any test that calls the method?  Maybe have a line after that checks the non-upsert fields against the output file again and asserts they are the same?  That way we could go back and update base whenever it changes.  What are your thoughts?

Reply:
get_upsert_only_fields has hard-coded non-upsert fields for streams and returns the difference from all the fields as selected fields. There is not a lot happening there. The next line, we are calling the super class test method which uses this selected_fields against the actual fields. This is the actual test and if there is any new field or added, it will be flagged as an error from the assertion. I don't see any risk here.

Reply:
Thanks I think I see it now.  So if a field moves from non-upsert to upsert it will be missing from expected_all_keys but present in fields_replicated and cause a test failure.  Similarly if a field moves from upsert to non-upsert it will be missing from fields_replicated but still present in expected_all_keys and cause a test failure.